internal/client/datadog/formatter_test.go

```diff
@@ -102,39 +104,15 @@ var _ = Describe("Formatter", func() {
 		Expect(s.Host).To(Equal("some.host"))
 		Expect(s.Tags).To(Equal([]string{"some:tag", "other:tag"}))
 
-		// now test a scenario where we're actually splitting points
-		m = make(map[metric.MetricKey]metric.MetricValue)
-		m[metric.MetricKey{Name: "a"}] = metric.MetricValue{
-			Points: []metric.Point{{Value: 9}, {Value: 10}},
-			Tags:   []string{"some:tag", "other:tag"},
-			Host:   "some.host",
+		// now test a scenario where we're actually splitting metrics
+		for i := 0; i < 1000; i++ {
+			k, v := makeFakeMetric("metricName", 1000, uint64(i), defaultTags)
+			metricsMap.Add(k, v)
 		}
-		result = formatter.Format("some-prefix.", 1, m)
-
-		Expect(result).To(HaveLen(2))
-
-		decompressed1 := helper.Decompress(result[0])
-		payload1 := Payload{}
-		err1 := json.Unmarshal(decompressed1, &payload1)
-		Expect(err1).To(BeNil())
-		Expect(payload1.Series).To(HaveLen(1))
-		s1 := payload1.Series[0]
-		Expect(s1.Metric).To(Equal("some-prefix.a"))
-		Expect(s1.Points).To(Equal([]metric.Point{{Value: 9}}))
-		Expect(s1.Type).To(Equal("gauge"))
-		Expect(s1.Host).To(Equal("some.host"))
-		Expect(s1.Tags).To(Equal([]string{"some:tag", "other:tag"}))
-
-		decompressed2 := helper.Decompress(result[1])
-		payload2 := Payload{}
-		err2 := json.Unmarshal(decompressed2, &payload2)
-		Expect(err2).To(BeNil())
-		Expect(payload2.Series).To(HaveLen(1))
-		s2 := payload2.Series[0]
-		Expect(s2.Metric).To(Equal("some-prefix.a"))
-		Expect(s2.Points).To(Equal([]metric.Point{{Value: 10}}))
-		Expect(s2.Type).To(Equal("gauge"))
-		Expect(s2.Host).To(Equal("some.host"))
-		Expect(s2.Tags).To(Equal([]string{"some:tag", "other:tag"}))
+
+		a, b := splitMetrics(metricsMap)
+
+		Expect(len(a)).To(BeNumerically("<=", len(metricsMap)))
+		Expect(len(b)).To(BeNumerically("<=", len(metricsMap)))
```

Comment:
you're not really testing anything here. did you mean 
```suggestion
		Expect(len(a)).To(BeNumerically("<=", len(metricsMap)/2))
		Expect(len(b)).To(BeNumerically("<=", len(metricsMap)/2))
```

Reply:
I even think you can assert for the exact length: you know how many metrics you have and how they're split, so you know how many metrics should be in `a` and in `b`

Reply:
Yes 🤦